File: java/src/Provider/PaymentProvider.java
package Provider;

// Provedor do serviço de pagamento — expõe a lógica de negócio como serviço
public class PaymentProvider {

    public String processPayment() {
        return processPayment(100.0, "cliente-padrão");
    }

    public String processPayment(double amount, String clientId) {
        System.out.println("  [PaymentProvider] Processando pagamento de R$ " + amount + " para " + clientId);
        return "Pagamento de R$ " + amount + " processado com sucesso para o cliente: " + clientId;
    }
}

File: python/provider/payment_provider.py
# Provedor do serviço de pagamento — expõe a lógica de negócio como serviço
class PaymentProvider:
    def process_payment(self, amount: float = 100.0, client_id: str = "cliente-padrão") -> str:
        print(f"  [PaymentProvider] Processando pagamento de R$ {amount} para {client_id}")
        return f"Pagamento de R$ {amount} processado com sucesso para o cliente: {client_id}"
